lib/rules/no-wait-for-snapshot.ts

```diff
@@ -0,0 +1,115 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils';
+import { getDocsUrl, ASYNC_UTILS, LIBRARY_MODULES } from '../utils';
+import { findClosestCallExpressionNode } from '../node-utils';
+
+export const RULE_NAME = 'no-wait-for-snapshot';
+export type MessageIds = 'noWaitForSnapshot';
+type Options = [];
+
+const ASYNC_UTILS_REGEXP = new RegExp(`^(${ASYNC_UTILS.join('|')})$`);
+
+export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Ensures no snapshot is generated inside of a `waitFor` call',
+      category: 'Best Practices',
+      recommended: 'warn',
+    },
+    messages: {
+      noWaitForSnapshot:
+        "A snapshot can't be generated inside of a `waitFor` call",
```

Comment:
I think it'd be better to here use the async utility message that caused the error, rather than the hardcoded `waitFor`.

Otherwise, we'd face scenarios like the following:

```ts
await wait(() => {
    expect(container).toMatchSnapshot();
  });
// output error message: 'Ensures no snapshot is generated inside of a `waitFor` call'
// expected output error message: 'Ensures no snapshot is generated inside of a `wait` call'
```

Reply:
Good call